gnat2goto/driver/tree_walk.adb

```diff
@@ -1433,9 +1433,13 @@ package body Tree_Walk is
       -- Do_Pragma_Assert --
       ----------------------
 
-      procedure Do_Pragma_Assert (N : Node_Id; Block : Irep);
-      procedure Do_Pragma_Assert (N : Node_Id; Block : Irep) is
-         Assert_Irep : constant Irep := New_Irep (I_Code_Assert);
+      procedure Do_Pragma_Assert_or_Assume
+        (N : Node_Id; Block : Irep; Which : String);
+      procedure Do_Pragma_Assert_or_Assume
+        (N : Node_Id; Block : Irep; Which : String) is
+         A_Irep : constant Irep := (if Which = "assert"
```

Comment:
here you can use if-expression just for the argument

```... := New_Irap ((if Which = ... then ... else ...))```

that's mostly a matter of style, so feel free to leave as it is (though the more concise version is likely to be compiled into fewer assembly instructions; I checked this some time ago)